packages/eslint-plugin/tests/rules/no-unsafe-unary-minus.test.ts

```diff
@@ -0,0 +1,47 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-unsafe-unary-minus';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('no-unsafe-unary-minus', rule, {
+  valid: [
+    '+42;',
+    '-42;',
+    '-42n;',
+    '(a: number) => -a;',
+    '(a: bigint) => -a;',
+    '(a: number | bigint) => -a;',
+    '(a: any) => -a;',
+    '(a: 1 | 2) => -a;',
+    '(a: string) => +a;',
+    '(a: number[]) => -a[0];',
+    '<T,>(t: T & number) => -t;',
+    '(a: { x: number }) => -a.x;',
+    '(a: never) => -a;',
```

Comment:
[Functionality] Interesting. What use would there be for `-a`? I suspect this should be an error, but am not sure...

Reply:
No, the `never` type is supposed to have no elements, to any operation on it is valid.

Reply:
But why would someone `-` it? To me, that's more likely a sign of a mistake: that they thought it had a value but it actually doesn't.

Reply:
Sure, but that's entirely orthogonal: if we want to make a rule saying not to use a `never` type in a context that allows a broader type, then we could (although it seems a bit weird), but it is unrelated to this rule about `-`.

Reply:
cc @typescript-eslint/triage-team - I'm of the opinion this rule would make sense for that. Anybody have thoughts?

Reply:
To clarify my position: I'm actually pretty strongly opposed to disallowing `never` in this rule. The whole point of `never` is that it serves as a [bottom type](https://en.wikipedia.org/wiki/Bottom_type), and it arises naturally when you take the intersection of two disjoint types. It would be inconsistent and surprising for it to be disallowed here.

Reply:
I don't see why forbidding it either—it's mathematically sound. Our heuristic is basically "allow only operands assignable to `number`/`bigint`" and `never` fits.